src/java/com/palantir/cassandra/check/VolumeIntegrityCheck.java

```diff
@@ -0,0 +1,114 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.cassandra.check;
+
+import java.io.IOException;
+import java.nio.file.Paths;
+import java.util.Arrays;
+import java.util.Optional;
+import java.util.UUID;
+
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.palantir.cassandra.utils.FileParser;
+import com.palantir.logsafe.Preconditions;
+import org.apache.cassandra.config.DatabaseDescriptor;
+
+/**
+ * Check if the backing volume has been tampered with. While Cassandra supports multiple data drive, we only
+ * use 1 data drive.
+ */
+public final class VolumeIntegrityCheck
+{
+    public static final String METADATA_NAME = "cassandra-metadata.json";
+
+    private final UUID hostId;
+
+    private final FileParser<VolumeMetadata> dataDriveMetadataFileParser;
+
+    private final FileParser<VolumeMetadata> commitLogMetadataFileParser;
+
+    static class VolumeMetadataType extends TypeReference<VolumeMetadata>
+    {
+    }
+
+    public VolumeIntegrityCheck(UUID hostId,
+                                FileParser<VolumeMetadata> dataDriveMetadataFileParser,
+                                FileParser<VolumeMetadata> commitLogMetadataFileParser)
+    {
+        this.hostId = hostId;
+        this.dataDriveMetadataFileParser = dataDriveMetadataFileParser;
+        this.commitLogMetadataFileParser = commitLogMetadataFileParser;
+    }
+
+    public void execute()
+    {
+        try
+        {
+            Optional<VolumeMetadata> maybeDataDriveMetadata = dataDriveMetadataFileParser.read();
+            Optional<VolumeMetadata> maybeCommitLogMetadata = commitLogMetadataFileParser.read();
+            check(maybeDataDriveMetadata, maybeCommitLogMetadata);
+
+            // Write order matters here. We treat write to the data drive as
+            // a "locking" operation, i.e the node is in a steady state.
```

Comment:
What does "steady state" mean here?

Reply:
Steady state means that the changes have been rolled onto existing nodes. I wanted to specify the distinction that a given node may be waiting to bootstrap vs a node that existed prior to this change. Without writing to data drive, we won't be able to distinguish between the two cases.

Reply:
Maybe I can include ^ blurb to make it more clear.